java/videostore/src/Customer.java

```diff
@@ -1,67 +1,81 @@
-
-		while (rentals.hasMoreElements ()) {
-			double 		thisAmount = 0;
-			Rental 		each = (Rental)rentals.nextElement ();
-
-			// determines the amount for each line
-			switch (each.getMovie ().getPriceCode ()) {
-				case Movie.REGULAR:
-					thisAmount += 2;
-					if (each.getDaysRented () > 2)
-						thisAmount += (each.getDaysRented () - 2) * 1.5;
-					break;
-				case Movie.NEW_RELEASE:
-					thisAmount += each.getDaysRented () * 3;
-					break;
-				case Movie.CHILDRENS:
-					thisAmount += 1.5;
-					if (each.getDaysRented () > 3)
-						thisAmount += (each.getDaysRented () - 3) * 1.5;
-					break;
-			}
-
-			frequentRenterPoints++;
-
-			if (each.getMovie ().getPriceCode () == Movie.NEW_RELEASE
-					&& each.getDaysRented () > 1)
-				frequentRenterPoints++;
-
-			result += "\t" + each.getMovie ().getTitle () + "\t"
-								+ String.valueOf (thisAmount) + "\n";
-			totalAmount += thisAmount;
-
-		}
-
-		result += "You owed " + String.valueOf (totalAmount) + "\n";
-		result += "You earned " + String.valueOf (frequentRenterPoints) + " frequent renter points\n";
-
-
-		return result;
-	}
-
-
-	private String name;
-	private Vector rentals = new Vector ();
+import java.util.ArrayList;
+import java.util.List;
+
+public class Customer {
+    private String name;
+    private List<Rental> rentals = new ArrayList<>();
+    private double amountOwed;
+    private int frequentRenterPoints;
+
+    public Customer(String name) {
+        this.name = name;
+    }
+
+    public void addRental(Rental rental) {
+        rentals.add(rental);
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String statement() {
+        amountOwed = 0;
+        frequentRenterPoints = 0;
+        String result = "Rental Record for " + getName() + "\n";
+
+        for (Rental eachMovie : rentals) {
+            double currentAmount = 0;
+
+            amountOwed += getAmountAccrued(eachMovie, currentAmount);
+
+            getFrequentRenterPoints(eachMovie);
+
+            result += "\t" + eachMovie.getMovie().getTitle() + "\t" + String.valueOf(currentAmount) + "\n";
+        }
+
+        result += "You owed " + String.valueOf(amountOwed) + "\n";
+        result += "You earned " + String.valueOf(frequentRenterPoints) + " frequent renter points\n";
+
+
+        return result;
+    }
+
+    public double getAmountOwed() {
+        return amountOwed;
+    }
+
+    public int getFrequentRenterPoints() {
+        return frequentRenterPoints;
+    }
+
+    private double getAmountAccrued(Rental eachMovie, double currentAmount) {
+        return amountAccruedForRentals(currentAmount, eachMovie);
+    }
+    private void getFrequentRenterPoints(Rental eachMovie) {
+        frequentRenterPoints++;
+
+        if (eachMovie.getMovie().getPriceCode() == Movie.NEW_RELEASE
+                && eachMovie.getDaysRented() > 1)
+            frequentRenterPoints++;
+    }
+
+    private double amountAccruedForRentals(double currentAmount, Rental each) {

```

Comment:
`each` is not very descriptive/informative. The effect of naming the rental variable `each` is more obvious when you look at this: `each.getMovie()`
- What does `each` mean?
- What kind of `each` has a movie attached to it?

The above questions are easily answered when you name the variable to what it is `rental`. The problematic line then changes to `rental.getMovie()` which tells me all that I need to know.